packages/graphics/src/Graphics.js

```diff
@@ -829,143 +829,198 @@ export default class Graphics extends Container
 
-            const uniforms = this.shader.uniforms;
+        // this means that we can tack advantage of the sync function of pixi!
+        // bind and sync uniforms..
+        // there is a way to optimise this..
+        renderer.shader.bind(shader);
+        renderer.geometry.bind(geometry, shader);
 
-            // lets set the transfomr
-            uniforms.translationMatrix = this.transform.worldTransform;
+        // set state..
+        renderer.state.set(this.state);
 
-            const tint = this.tint;
-            const wa = this.worldAlpha;
+        // then render the rest of them...
+        for (let i = 0, l = drawCalls.length; i < l; i++)
+        {
+            this._renderDrawCallDirect(renderer, geometry.drawCalls[i]);
+        }
+    }
 
-            // and then lets set the tint..
-            uniforms.tint[0] = (((tint >> 16) & 0xFF) / 255) * wa;
-            uniforms.tint[1] = (((tint >> 8) & 0xFF) / 255) * wa;
-            uniforms.tint[2] = ((tint & 0xFF) / 255) * wa;
-            uniforms.tint[3] = wa;
+    /**
+     * Renders specific DrawCall
+     *
+     * @param {PIXI.Renderer} renderer
+     * @param {PIXI.BatchDrawCall} drawCall
+     */
+    _renderDrawCallDirect(renderer, drawCall)
+    {
+        const groupTextureCount = drawCall.textureCount;
 
-            // the first draw call, we can set the uniforms of the shader directly here.
+        for (let j = 0; j < groupTextureCount; j++)
+        {
+            renderer.texture.bind(drawCall.textures[j], j);
+        }
 
-            // this means that we can tack advantage of the sync function of pixi!
-            // bind and sync uniforms..
-            // there is a way to optimise this..
-            renderer.shader.bind(this.shader);
+        renderer.geometry.draw(drawCall.type, drawCall.size, drawCall.start);
+    }
 
-            // then render it
-            renderer.geometry.bind(geometry, this.shader);
+    /**
+     * Resolves shader for direct rendering
+     *
+     * @protected
+     * @param {PIXI.Renderer} renderer - The renderer
+     */
+    _resolveDirectShader(renderer)
+    {
+        let shader = this.shader;
 
-            // set state..
-            renderer.state.set(this.state);
+        const pluginName = this.pluginName;
 
-            // then render the rest of them...
-            for (let i = 0; i < geometry.drawCalls.length; i++)
+        if (!shader)
+        {
+            // if there is no shader here, we can use the default shader.
+            // and that only gets created if we actually need it..
+            // but may be more than one plugins for graphics
+            if (!DEFAULT_SHADERS[pluginName])
             {
-                const drawCall = geometry.drawCalls[i];
-
-                const groupTextureCount = drawCall.textureCount;
+                const sampleValues = new Int32Array(16);
 
-                for (let j = 0; j < groupTextureCount; j++)
+                for (let i = 0; i < 16; i++)
```

Comment:
Hardcode 16 here will break some Adreno 3xx GPU devices.
Because sampler array may be less than 16, but you set a Int32Array with 16 length to the sampler array uniform  which will cause GPU driver crash.

Reply:
Probably better to open an issue now.

Reply:
PR welcome to fix.

Reply:
Here it is: https://github.com/pixijs/pixi.js/pull/6876